### PR TITLE
Fix/loading 404

### PR DESCRIPTION
**Pull request by jordaniza**
NOTE: this fix builds off the [PR for api-v2](https://github.com/BanklessDAO/bounty-board/pull/163), and is intended to be pushed to develop after that branch.

This fix removes the pop in showing a 404 page whilst a bounty is loading.

While investigating the fix, I noticed a significant number of re-renders on the main bounties page leading to several unneccessary calls. What we then did was:

- Split the bounty accordion from the single bounty page, instead of conditionally rendering based on whether an `id` param is passed
- Extract some of the data fetching logic into hooks
- Remove conditional logic in the bounties page, as it now will _only_ render an array
- Added unit tests for the frontend single  bounty page in jest/enzyme.

### CODE REVIEW

File: packages/react-app/src/hooks/useBounties.ts [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+export function useBounties(url: string): useBountiesResponse {
+	/**
+   * Wraps the SWR hook with additional loading state for spinners
+   */
+	const { data, error } = useSWR<BountyCollection[], unknown>(url, axiosFetcher);
+	return {
+		bounties: data,
+		isLoading: !error && !data,
+		isError: !!error,
```
Comment (Behold3th) [inline packages/react-app/src/hooks/useBounties.ts:29]: `!!error` vs `error` ?

File: packages/react-app/src/models/Bounty.ts [hunk 2]
```diff
@@ -7,6 +7,10 @@ import {
 	array,
 	mixed,
 } from 'yup';
+
+// Have to use ignore here due to some strange error with typescript not picking
+// up the declaration file, unless you change the name every deploy
+// @ts-ignore
```
Comment (Behold3th) [inline packages/react-app/src/models/Bounty.ts:13]: I thought the `ban-ts-annotation` config set earlier in this PR would disallow this?

https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/eslint-plugin/docs/rules/ban-ts-comment.md

Comment (jordaniza) [reply packages/react-app/src/models/Bounty.ts:13]: Disabled the setting.

File: packages/react-app/src/hooks/useBounties.ts (hunk 1 above)
Comment (jordaniza) [reply packages/react-app/src/hooks/useBounties.ts:29]: !!error coalesces the error value as a boolean, it is equivalent to Boolean(error)

File: packages/react-app/src/pages/[id]/index.tsx [hunk 3]
```diff
@@ -1,21 +1,66 @@
...
+				<ColorModeButton>Go Back</ColorModeButton>
+			</Box>
+		</Link>
+	</Stack>
+);
 
-const BountyPage = (): JSX.Element => {
+export const BountyLoader = () => (
+	<Box padding='6' boxShadow='lg'>
+		<SkeletonCircle size='10' />
```
Comment (Behold3th) [inline packages/react-app/src/pages/[id]/index.tsx:29]: https://github.com/BanklessDAO/bounty-board/issues/178

**Review by Behold3th — APPROVED** [review]
🔥 work, @jordaniza !

**Pull request merged**